### PR TITLE
set_solder_ball enhancement

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

File: pyaedt/edb_core/components.py [hunk 1]
```diff
@@ -1287,20 +1287,28 @@ def disable_rlc_component(self, component_name):
         return False
 
     @pyaedt_function_handler()
-    def set_solder_ball(self, component="", sball_diam="100um", sball_height="150um"):
+    def set_solder_ball(
+        self, component="", shape="Cylinder", sball_diam="100um", sball_height="150um", sball_mid_diam=None
+    ):
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:1292]: @ring630 I would just add the new parameters at the end.
```suggestion
    def set_solder_ball(
        self, component="", sball_diam="100um", sball_height="150um", shape="Cylinder", sball_mid_diam=None
    ):
```

File: pyaedt/edb_core/components.py [hunk 2]
```diff
@@ -1287,20 +1287,28 @@ def disable_rlc_component(self, component_name):
...
+            ``"Spheroid``. The default is ``"Cylinder"``.
+
         sball_diam  : str, float, optional
             Diameter of the solder ball.
 
         sball_height : str, float, optional
             Height of the solder ball.
 
+        sball_mid_diam: str, float, optional
+            Mid diamter of the solder ball.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:1311]:
```suggestion
        Parameters
        ----------
        component_name : str or EDB component
            Name of the discrete component.
        sball_diam  : str, float, optional
            Diameter of the solder ball.
        sball_height : str, float, optional
            Height of the solder ball.
        shape: str, optional
            Shape of solder ball. Options are ``"Cylinder"``,
            ``"Spheroid"``. The default is ``"Cylinder"``.
        sball_mid_diam: str, float, optional
            Mid diameter of the solder ball.
```

File: pyaedt/edb_core/components.py (hunk 1 above)
Comment (ring630) [reply pyaedt/edb_core/components.py:1292]: Thanks for the comment.

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Comment by MaxJPRey** [comment]
@ring630 well done for the refactoring of the method and removing some "copy/paste" by some global logic.

**Pull request merged**